### PR TITLE
Fix firmware upload on Windows

### LINKED ISSUE #3731: [3.3.1] Can't upload firmware
**Application Version**
3.3.1

**Platform**
Windows 10

**Display Driver**

**Printer**
Custom Prusa MK2S DIY

**Steps to Reproduce**
Select printer -> manage printer -> upload firmware -> "select and confirm"

**Actual Results**
Uploading window appears but no progress

**Expected results**
Upload firmware

**Additional Information**
On 3.2 version it was working

### PR DESCRIPTION
**Pull request by fieldOfView**
This PR fixes firmware uploading on Windows by correctly creating a local path from a url-encoded path. 

On Windows, local file URLs start with file:/// (three forwards slashes). Instead of hardcoding a fix for Windows only, this PR uses QUrl to handle this properly.

Fixes #3731 and #3987

### CODE REVIEW

**Comment by fieldOfView** [comment]
Further testing showed that the default firmware (if any, for the selected printer) uploaded ok, but a custom firmware did not upload. That's because the default firmware location is supplied as a local file, whereas the custom firmware location is supplied QUrl encoded. Both are now handled correctly.

**Comment by fieldOfView** [comment]
subtle bump
This has no Jira issue yet.

**Comment by ChrisTerBeke** [comment]
Devs: CURA-5588 :)

File: plugins/USBPrinting/USBPrinterOutputDevice.py [hunk 1]
```diff
@@ -127,8 +127,11 @@ def showFirmwareInterface(self):
 
     @pyqtSlot(str)
     def updateFirmware(self, file):
-        # the file path is qurl encoded.
-        self._firmware_location = file.replace("file://", "")
+        # the file path could be qurl encoded.
+        if file[:7] == "file://":
```
Comment (LipuFei) [inline plugins/USBPrinting/USBPrinterOutputDevice.py]: Minor: I think `file.startswith("file://")` is better.

Comment (fieldOfView) [reply plugins/USBPrinting/USBPrinterOutputDevice.py]: Fixed: 846c55a99d0486bea3623e47412c36d0ab71f0fe

**Pull request merged**